ceph/ceph_admin/osd.py

```diff
@@ -106,3 +106,106 @@ def apply(self, config: Dict) -> None:
+            op: operation parameters ex: remove
+        Example:
+            Testing ceph orch osd rm status
+        config:
+          command: rm
+
+        Returns:
+          output, error   returned by the command.
+
+        example
+            config:
+                command: rm status
+                base_cmd_args:
+                    verbose: true
+                args:
+                    format: json-pretty
+
+        """
+
+        base_cmd = ["ceph", "orch", "osd", "rm", "status"]
+        if config.get("base_cmd_args"):
+            base_cmd.append(config_dict_to_string(config["base_cmd_args"]))
+
+        if config and config.get("args"):
+            args = config.get("args")
+            base_cmd.append(config_dict_to_string(args))
+
+        return self.shell(args=base_cmd)
+
+    def rm(self, config):
+        """
+        Execute the command ceph orch osd rm <OSD ID> .
+        Args:
+
+            config: command and service are passed from the test case.
+            op: operation parameters ex: remove
+        Example:
+            Testing ceph orch osd rm 1
+        config:
+          command: rm
+
+        Returns:
+          output, error   returned by the command.
+
+        example
+            config:
+                command: rm
+                base_cmd_args:
+                    verbose: true
+                pos_args:
+                    - 1
+
+        """
+
+        base_cmd = ["ceph", "orch", "osd"]
+        if config.get("base_cmd_args"):
+            base_cmd.append(config_dict_to_string(config["base_cmd_args"]))
+        base_cmd.append("rm")
+        osd_id = str(config["pos_args"][0])
+        base_cmd.append(osd_id)
+        self.shell(args=base_cmd)
+
+        check_osd_id_dict = {
+            "args": {"format": "json-pretty"},
+        }
+
+        while True:
+            # "ceph orch osd rm status -f json-pretty"
+            # condition
+            # continue loop if OSD_ID present
+            # if not exit the loop
+            out, _ = self.rm_status(check_osd_id_dict)
+
+            try:
+                status = json.loads(out)
+                for osd_id_ in status:
+                    if osd_id_["osd_id"] == osd_id:
+                        LOG.info(
+                            f"OSDs are still up as Draining in progress: {osd_id_}"
+                        )
+                        sleep(2)
+                        continue
+
+                else:
+                    break
+            except json.decoder.JSONDecodeError:
+                break
+
+        # validate OSD removal
+        out, verify = self.shell(
+            args=["ceph", "osd", "tree", "-f", "json-pretty"],
+        )
+        out = json.loads(out)
+        for id_ in out["nodes"]:
+            if id_["id"] == osd_id:
+                raise AssertionError("fail, OSD is present still after removing")
+                LOG.info("OSD removed Id not present, pass")
```

Comment:
Hope this code is unreachable, Since exception is raised before logging.

Reply:
I think this should be,
` LOG.error("OSD Removed Id Found...")`

Reply:
Done